src/transformers/models/mplug_owl/modeling_mplug_owl.py

```diff
@@ -0,0 +1,1574 @@
+        self.config = config
+        self.layers = nn.ModuleList(
+            [MplugOwlVisualAbstractorLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
+        )
+        self.gradient_checkpointing = False
+
+    def forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        head_mask=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        past_key_values=None,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=True,
+    ):
+        all_hidden_states = () if output_hidden_states else None
+
+        for i in range(self.config.num_hidden_layers):
+            layer_module = self.layers[i]
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            layer_head_mask = head_mask[i] if head_mask is not None else None
+            past_key_value = past_key_values[i] if past_key_values is not None else None
+
+            if getattr(self.config, "gradient_checkpointing", False) and self.training:
+
+                def create_custom_forward(module):
+                    def custom_forward(*inputs):
+                        return module(*inputs, past_key_value, output_attentions)
+
+                    return custom_forward
+
+                layer_outputs = torch.utils.checkpoint.checkpoint(
+                    create_custom_forward(layer_module),
+                    hidden_states,
+                    attention_mask,
+                    layer_head_mask,
+                    encoder_hidden_states,
+                    encoder_attention_mask,
+                )
+            else:
+                layer_outputs = layer_module(
+                    hidden_states,
+                    attention_mask,
+                    layer_head_mask,
+                    encoder_hidden_states,
+                    encoder_attention_mask,
+                    output_attentions,
+                )
+
+            hidden_states = layer_outputs[0]
+
+        return BaseModelOutput(
+            last_hidden_state=hidden_states,
+        )
+
+
+class MplugOwlVisualAbstractorModel(MplugOwlPreTrainedModel):
+    """
+    Args:
+    MplugOwlVisualAbstractorModel
+        config (`MplugOwlVisualAbstractorConfig`):
+            The corresponding configuration for the `MplugOwlVisualAbstractorModel`.
+    """
+
+    def __init__(self, config: MplugOwlVisualAbstractorConfig, language_hidden_size):
+        super().__init__(config)
+        self.config = config
+
+        self.encoder = MplugOwlVisualAbstractorEncoder(config)
+        self.visual_fc = torch.nn.Linear(config.hidden_size, language_hidden_size)
+        self.vit_eos = torch.nn.Parameter(torch.randn(1, 1, language_hidden_size))
+        self.post_init()
+
+    def _init_weights(self, module):
+        super()._init_weights(module)
+        nn.init.trunc_normal_(self.vit_eos, mean=0.0, std=self.config.initializer_range)
```

Comment:
Can you add this method inside `_init_weights` of `MplugOwlVisualPreTrainedModel` with a check that checks for the correct class?

```python
if isinstance(module, MplugOwlVisualAbstractorModel):
    nn.init.trunc_normal_(module.vit_eos, mean=0.0, std=self.config.initializer_range)